Tests/MongoSwiftBenchmarks/BenchmarkUtils.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+import XCTest
+
+extension XCTestCase {
+
+    /// Measure the time for a single execution of operation
+    func measureTime(_ operation: () throws -> Void) throws -> Double {
+        let startTime = ProcessInfo.processInfo.systemUptime
+        try operation()
+        let timeElapsed = ProcessInfo.processInfo.systemUptime - startTime
+        return Double(timeElapsed)
+    }
+
+    /// Measure the median time for iterations executions of operation
+    func measureOp(_ operation: () throws -> Void, iterations: Int = 100) throws -> Double {
+        var results = [Double]()
+        for _ in 1...iterations {
+            results.append(try measureTime(operation))
+        }
+        return median(results)
+    }
+
+    /// Compute the median of an array of doubles 
+    func median(_ input: [Double]) -> Double {
+        return input.sorted(by: <)[input.count / 2]
+    }
+
+    func printResults(time: Double, size: Double) {
+        let roundedTime = Double(floor(time * 1000) / 1000)
+        let roundedScore = Double(floor(size / time * 10000) / 10000)
+        print("Results for \(self.name): median time \(roundedTime) seconds, score \(roundedScore) MB/s")
```

Comment:
If you're already printing this out do we even need the python script? Can't we just print out JSON right here?

Reply:
so [here's the script](https://github.com/mongodb/mongo-python-driver/blob/master/.evergreen/run-perf-tests.sh) the Python driver uses for running perf tests on Evergreen, and [here's](https://github.com/mongodb/mongo-python-driver/blob/master/test/performance/perf_test.py) the actual `perf_test.py` file. it looks like they keep this global `result_data` list and at the end dump that all out to a `results.json` file. 

then, looking at their [perf project](https://evergreen.mongodb.com/task/mongo_python_driver_perf_perf_tests_perf_3.0_standalone_f847be3fcb1c75801576748128b0440de144304f_18_03_22_15_47_34) (see log at bottom) they do some `attach.results` and `json.send` commands using that file. I'm not sure how that evergreen magic works, looks like there are related commands defined in their [`config.yml`](https://github.com/mongodb/mongo-python-driver/blob/master/.evergreen/config.yml#L341) and [`perf.yml`](https://github.com/mongodb/mongo-python-driver/blob/15f62db70c4e9e81f9fdd8075df5f6d71143c23c/.evergreen/perf.yml#L130).

SO, based on all that, I've been operating under the assumption that we need to output all the results together at the end in a single file, meaning that we can't just individually print out JSON results interspersed with the other test output and have evergreen understand it.

so it seems to me like we do need the python script, and probably what should be happening in there is that (rather than just printing `benchmarks` as I do now) we dump it out to a JSON file which we then somehow tell evergreen to use for plots.

Reply:
well we don't want to force developers to have a random file on the system each time they run the benchmark tests, I think what you have here is actually probably fine, I was just wondering why we needed the extra step

Reply:
well, you could just run `make benchmark` and not pipe it if you don't want this stuff, but regardless it's more flexible/less work to just pipe `benchmark.py`'s output to another file so ya let's just leave it as printing. 
I guess we should really be printing `json.dumps(benchmarks)` though so it's properly formatted, but I think I'll have to do some more finagling with the output structure to make evergreen understand it anyway, so we can worry about that when we actually have automated running the benchmarks.
we should think about what exactly we want to plot - python does ops/second, or we could just plot the times or scores, or I guess we can probably plot multiple of them